### PR TITLE
Add a vegan diversity index tool.

### PR DESCRIPTION
**Pull request by blankenberg**

### CODE REVIEW

**Comment by bgruening** [comment]
@blankenberg if we keep adding new tools here can we change https://github.com/galaxyproject/tools-devteam/blob/master/CONTRIBUTING.md

The vegan suite is quite new and I don't see `exceptional conditions` here. So we should remove this section, right?
Do you plan to add more tools? If so it might be good to share a macro file and reorganise the folder structure.

File: tools/vegan_diversity/vegan_diversity.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        </configfile>
+    </configfiles>
+    <inputs>
+        <param name="input_abundance" type="data" format="tabular" label="File with abundance values for community" help="Rows are samples; columns are species/phyla/community classifier"/>
+        <param name="species_column" label="Group name column" type="data_column" data_ref="input_abundance" value="6" help="Species, phylum, etc"/>
+        <param name="sample_columns" label="Sample count columns" type="data_column" multiple="True" value="2" data_ref="input_abundance" help="Select each column that contains counts"/>
+        <param name="header" type="boolean" truevalue="TRUE" falsevalue="FALSE" checked="False" label="Input has a header line"/>
+        <conditional name="index_type">
+            <param name="index_type_selector" type="select" label="Diversity index to use">
+                <option value="shannon" selected="True">shannon</option>
```
Comment (bgruening) [inline tools/vegan_diversity/vegan_diversity.xml]: Shannon-Weaver

File: tools/vegan_diversity/vegan_diversity.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                <option value="simpson">simpson</option>
+                <option value="invsimpson">invsimpson</option>
```
Comment (bgruening) [inline tools/vegan_diversity/vegan_diversity.xml]: inverse Simpson

File: tools/vegan_diversity/vegan_diversity.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            <when value="shannon">
+                <param name="margin" type="integer" value="1" label="Margin for which the index is computed"/>
+                <param name="base" type="select" label="The logarithm base used in shannon">
+                    <option value="exp(1)" selected="True">exp(1)</option>
+                    <option value="2" selected="False">2</option>
+                </param>
+            </when>
+            <when value="simpson">
+                <param name="margin" type="integer" value="1" label="Margin for which the index is computed"/>
+                <param name="base" type="select" label="The logarithm base used in shannon">
```
Comment (bgruening) [inline tools/vegan_diversity/vegan_diversity.xml]: The label is wrong in 2/3 cases :)

File: tools/vegan_diversity/vegan_diversity.xml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+                    <option value="exp(1)" selected="True">exp(1)</option>
```
Comment (bgruening) [inline tools/vegan_diversity/vegan_diversity.xml]: Can we make the user visible part cleaner, exp(1)?

File: tools/vegan_diversity/vegan_diversity.xml [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            <param name="sample_columns" value="2"/>
+            <param name="index_type_selector" value="shannon"/>
+            <param name="margin" value="1"/>
+            <param name="base" value="exp(1)"/>
+            <output name="output_slope" ftype="tabular" file="output_diversity_index.tabular" />
+        </test>
+    </tests>
+    <help>
+        <![CDATA[
+Calculate Diversity index using vegan and selected method.
```
Comment (bgruening) [inline tools/vegan_diversity/vegan_diversity.xml]: More help, with hyperlinks and more stuff from https://cran.r-project.org/web/packages/vegan/vignettes/diversity-vegan.pdf + a link to the vignette? Can we ship the vignettes like an image with the tool?

**Comment by blankenberg** [comment]
@bgruening vegan tools have been moved to tools-iuc if https://github.com/galaxyproject/tools-iuc/pull/258 is accepted.

**Comment by bgruening** [comment]
Thanks Dan!

**Pull request closed without merging**